### ISSUE TITLE
make:* commands fail on entities using Single Table Inheritance

### ISSUE THREAD

**Issue by IckleChris** [symfony/maker-bundle#146]
I have the following two entities in a Symfony 4.0.6 application (renamed to match the docs for the sake of this issue) using Doctrine's [Single Table Inheritance](http://docs.doctrine-project.org/projects/doctrine-orm/en/latest/reference/inheritance-mapping.html#single-table-inheritance) feature;

```
App\Entity\Person:
  type: entity
  table: person
  fields:
    id:
      type: integer
      id: true
      generator:
        strategy: AUTO
    name:
      type: string
      length: 250
      nullable: true
  inheritanceType: SINGLE_TABLE
  discriminatorColumn:
    name: person_type
    type: string
    length: 250
  discriminatorMap:
    thread: App\Entity\Employee
    reply: App\Entity\Manager
```

```
App\Entity\Employee:
  type: entity
  table: ~
  fields:
    ...
```

Running, for example `bin/console make:form EmployeeType Employee` presents the following error;

>In DoctrineEntityHelper.php line 67:
  Notice: Undefined offset: 0

The relevant code;

```
$metadata = $this->getEntityMetadata($entityClassName);

if (null !== $metadata) {
    return new DoctrineEntityDetails(
        $metadata->customRepositoryClassName,
        $metadata->identifier[0], // Error line
        $metadata->fieldMappings,
        $this->getFormFieldsFromEntity($metadata)
    );
}
```

Basically, because the entity metadata does not have an ID (as it exists in the parent) it errors. It doesn't look as though the `getIdentifier` method of `DoctrineEntityDetails` is in use anywhere anyway so I'm not sure what the identifier is required for. Temporarily replacing the line causing error results in the form being successfully created as expected. 

If the `make` features can't work without entities having an identifier, it may be worth adding an error to describe that. Thanks!

**Comment by javiereguiluz** [comment]
Thanks for reporting this issue. I haven't looked into the details of this ... but if fixing it would complicate our code a lot, I prefer to add this as a "non-supported edge case" of the bundle. Thanks!

**Comment by weaverryan** [comment]
@sadikoff what do you think? We should be able to fix this up :)

**Comment by sadikoff** [comment]
I'll try. I don't really like this part of code, and wanted to improve it!

**Comment by sadikoff** [comment]
@IckleChris Hello could you please provide more information about entities you are using? I've tried to reproduce bug with entities configured as in the doctrine docs, but unsuccessful :)
```php
// src/Entity/Person.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity(repositoryClass="App\Repository\PersonRepository")
 * @ORM\InheritanceType("SINGLE_TABLE")
 * @ORM\DiscriminatorColumn(name="person_type", type="string", length=250)
 * @ORM\DiscriminatorMap({"person" = "Person", "employee" = "Employee"})
 */
class Person
{
    /**
     * @ORM\Id()
     * @ORM\GeneratedValue()
     * @ORM\Column(type="integer")
     */
    protected $id;

    /**
     * @ORM\Column(type="string", length=255)
     */
    protected $name;

    public function getId()
    {
        return $this->id;
    }

    public function getName(): ?string
    {
        return $this->name;
    }

    public function setName(string $name): self
    {
        $this->name = $name;

        return $this;
    }
}
```
```php
// src/Entity/Employee
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity(repositoryClass="App\Repository\EmployeeRepository")
 */
class Employee extends Person
{

}
```

**Comment by IckleChris** [comment]
Hey @sadikoff 

I've put together a quick repo containing a slightly cut down version of my application's code, which allows the issue to be replicated - https://github.com/IckleChris/symfony-maker-146

For the sake of completeness;

1. Clone repo
2. `composer install`
3. Set up `.env` to configure DB connection
4. `bin/console doctrine:database:create`
5. `bin/console doctrine:migrations:migrate`
6. `bin/console make:form ThreadReportType ThreadReport`

Hope this helps :)

**Comment by sadikoff** [comment]
@IckleChris Perfect! This will help!

**Comment by MrNicodemuz** [comment]
@weaverryan I think this introduced a bug:

Undefined class constant 'ONE_TO_MANY'
